lockbox-ios/Store/UserInfoStore.swift

```diff
@@ -134,3 +135,53 @@ enum AutoLockSetting: String {
     case TwentyFourHours
     case Never
 }
+
+enum PreferredBrowserSetting: String {
+    case Chrome
+    case Firefox
+    case Focus
+    case Safari
+
+    static var defaultValue: PreferredBrowserSetting {
+        return PreferredBrowserSetting.Safari
+    }
+
+    func canOpenBrowser(application: UIApplication = UIApplication.shared) -> Bool {
+        var scheme: String?
+        switch self {
+        case .Safari:
+            return true
+        case .Firefox:
+            scheme = "firefox://open-url?url=http://mozilla.org"
+        case .Focus:
+            scheme = "firefox-focus://open-url?url=http://mozilla.org"
+        case .Chrome:
+            scheme = "googlechrome://"
+        }
+
+        if let scheme = scheme,
+            let url = URL(string: scheme) {
+            return application.canOpenURL(url)
+        }
+
+        return false
+    }
+
+    func openUrl(url: String, application: UIApplication = UIApplication.shared) {
+        var urlToOpen: URL?
+        switch self {
+        case .Safari:
+            urlToOpen = URL(string: "http://\(url)")
+        case .Firefox:
+            urlToOpen = URL(string: "firefox://open-url?url=http://\(url)")
```

Comment:
Have you tested that this works? I know there is some funny URL encoding required to pass full URLs to Firefox & Focus for iOS

Reply:
I tested but only with the simple domains that are in the app now. I'll fix it to encode the URL and re-test